### PR TITLE
Preset mastschild

### PR DESCRIPTION
**Pull request by bjoernma**
Füge Mastschilder zu JOSM-Preset für Hp, Ks, Hl-Haupt- und Kombinationssignale hinzu. Jetzt als zwei Eingabefelder, ich hoffe, das ist übersichtlich genug.

### CODE REVIEW

File: josm-presets/de-signals-eso.xml [hunk 1]
```diff
@@ -145,6 +145,22 @@ See https://wiki.openstreetmap.org/wiki/OpenRailwayMap for details.
 						display_values="substitute signal (Zs 1);cautioness signal (Zs 7);M Board (Zs 12)"
 						de.display_values="Ersatzsignal (Zs 1);Vorsichtsignal (Zs 7);M-Tafel (Zs 12)"
 						/>
+					<multiselect key="railway:signal:traversable:type"
+					    text="Traversable: action at stop"
+					    de.text="Mastschild: Verhalten bei Halt"
+					    values="written_default_speed;dictated_default_speed;always_on_sight;on_no_contact_on_sight"
+					    display_values="written_default_speed (traversable red/red-white/yellow-white);dictated_default_speed (traversable red/red-white);always_on_sight (traversable black-white);on_no_contact_on_sight (traversable yellow-white)"
+					    de.display_values="Mastschild rot / rot-weiß / gelb-weiß;Mastschild rot / rot-weiß;Mastschild schwarz-ẁeiß;Mastschild gelb-weiß"
```
Comment (DerDakon) [inline josm-presets/de-signals-eso.xml:153]: You have funny "ẁ" characters in every of these lines.

Comment (bjoernma) [reply josm-presets/de-signals-eso.xml]: I haven't really seen them, but they should be replaced. Adiotionally replaced some spaces through the vim ^I tab stop. Hope it's better now...

File: josm-presets/de-signals-eso.xml [hunk 2]
```diff
@@ -145,6 +145,22 @@ See https://wiki.openstreetmap.org/wiki/OpenRailwayMap for details.
 						display_values="substitute signal (Zs 1);cautioness signal (Zs 7);M Board (Zs 12)"
 						de.display_values="Ersatzsignal (Zs 1);Vorsichtsignal (Zs 7);M-Tafel (Zs 12)"
 						/>
+					<multiselect key="railway:signal:traversable:type"
+						text="Traversable: action at stop"
+						de.text="Mastschild: Verhalten bei Halt"
+						values="written_default_speed;dictated_default_speed;always_on_sight;on_no_contact_on_sight"
```
Comment (Nakaner) [inline josm-presets/de-signals-eso.xml:151]: I would replace the first multiselect box by one asking for the type of "Mastschild". It is easier for mappers to identify the board by their colour(s) than by its meaning (except those who have to follow their rules as their job ;-)).

What I mean:

```xml
delimiter="|"
values="written_default_speed;dictated_default_speed|on_no_contact_on_sight;written_default_speed;dictated_default_speed|always_on_sight|on_no_contact_on_sight"
display_values="white-red-white, red-triangle on white, or pure red|white-yellow-white-yellow-white|white-black-white-black-white"
de.display_values="weiß-rot-weiß, rotes Dreieck auf weiß oder nur rot|weiß-gelb-weiß-gelb-weiß|weiß-schwarz-weiß-schwarz-weiß"
```

**Review by Nakaner — CHANGES_REQUESTED** [review]
Could you please make use of a chunk as we do for the `ref=*` key`? It would reduce duplication and make the XML shorter.

File: josm-presets/de-signals-eso.xml (hunk 2 above)
Comment (bjoernma) [reply josm-presets/de-signals-eso.xml]: Should now be replaced and be rebased. Should work with the chunks. One disadvantage could be that it's not so easy to directly exclude non-existing combinations (s.a. Ks signals with white/yellow traversable should be quite rare...), but maybe better to use. Thanks!

File: styles/signals.mapcss [hunk 3]
```diff
@@ -2295,3 +2295,4 @@ node|z14-[railway=signal]["railway:signal:direction"]["railway:signal:brake_test
 	text-allow-overlap: true;
 	allow-overlap: true;
 }
+
```
Comment (DerDakon) [inline styles/signals.mapcss:2298]: Newline.

**Review by DerDakon — CHANGES_REQUESTED** [review]
When you are at it please squash.

**Review by DerDakon — APPROVED** [review]

File: josm-presets/de-signals-eso.xml [hunk 4]
```diff
@@ -38,6 +37,26 @@ See https://wiki.openstreetmap.org/wiki/OpenRailwayMap for details.
 			de.display_values="In Wegrichtung,Entgegen Wegrichtung,Beide Richtungen"
 			/>
 	</chunk>
+	<chunk id="traversable">
+		<multiselect key="railway:signal:traversable:type"
+			text="Traversable: action at stop"
+			de.text="Mastschild: Verhalten bei Halt"
+			delimiter="|"
```
Comment (Nakaner) [inline josm-presets/de-signals-eso.xml:44]: I tested your patch and found out that JOSM XXXXX is unable to handle the `|` delimiter properly.

error


However, a `#` works:

```diff
diff --git a/josm-presets/de-signals-eso.xml b/josm-presets/de-signals-eso.xml
index 93d69e0..42c4e46 100644
--- a/josm-presets/de-signals-eso.xml
+++ b/josm-presets/de-signals-eso.xml
@@ -41,10 +41,10 @@ See https://wiki.openstreetmap.org/wiki/OpenRailwayMap for details.
                <multiselect key="railway:signal:traversable:type"
                        text="Traversable: action at stop"
                        de.text="Mastschild: Verhalten bei Halt"
-                       delimiter="|"
-                       values="written_default_speed;dictated_default_speed|on_no_contact_on_sight;written_default_speed;dictated_default_speed|always_on_sight"
-                       display_values="white-red-white, red-triangle on white, or pure red|white-yellow-white-yellow-white|white-black-white-black-white"
-                       de.display_values="weiß-rot-weiß, rotes Dreieck auf weiß oder nur rot|weiß-gelb-weiß-gelb-weiß|weiß-schwarz-weiß-schwarz-weiß"
+                       delimiter="#"
+                       values="written_default_speed;dictated_default_speed#on_no_contact_on_sight;written_default_speed;dictated_default_speed#always_on_sight"
+                       display_values="white-red-white, red-triangle on white, or pure red#white-yellow-white-yellow-white#white-black-white-black-white"
+                       de.display_values="weiß-rot-weiß, rotes Dreieck auf weiß oder nur rot#weiß-gelb-weiß-gelb-weiß#weiß-schwarz-weiß-schwarz-weiß"
                        default="written_default_speed;dictated_default_speed"
                        />
                <combo key="railway:signal:traversable"
```

@DerDakon Should we work around this JOSM bug and merge the pull request? I can apply my bug fix patch during merge (separate commit).

**Review by Nakaner — CHANGES_REQUESTED** [review]

File: josm-presets/de-signals-eso.xml (hunk 4 above)
Comment (DerDakon) [reply josm-presets/de-signals-eso.xml:44]: You can't without tricks, you need to have in an approved branch first…

**Comment by DerDakon** [comment]
@bjoernma could you please rebase and change the delimiter as suggested by @Nakaner, then this can be merged.

File: josm-presets/de-signals-eso.xml (hunk 2 above)
Comment (DerDakon) [reply josm-presets/de-signals-eso.xml]: If there are combinations that do not exist then you can either add the specific combinations for that single signal type (if it's only one) or add another chunk.

**Comment by DerDakon** [comment]
You _may_ want to change the commit message of the first commit. I would only select the third of the 5 messages. You may also just squash them together. This is all your personal choice, I could as well just merge it.

**Review by Nakaner — APPROVED** [review]
You could remove the default for `railway:signal:traversable:type` but I am fine with it as it is now.

**Pull request merged**